### PR TITLE
Improves the copy of directories from groups to multigroups

### PR DESCRIPTION
**Pull request by Nicogp**
|Related issue|
|---|
|#8593|

## Description
This PR allows the copying of files from group subfolders when multigroups are created. Also adds validation to files to be shared to agents within group subfolders:

- Add the copy_directory function that recursively copies subdirectories
- Add function validate_shared_files which recursively validates files in directories and subdirectories
- Remove the recursion in the MergeAppendFile function, it became unnecessary
- We limit the maximum size of a group file path to 200 characters, due to the Windows limitation that file path size cannot exceed 260 characters. A margin of 60 characters was left because the 200 character limit does not include the characters corresponding to the installation path.

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
- [x] Source installation
- [x] Package installation
- [x] Source upgrade
- [x] Package upgrade
- [x] Review logs syntax and correct language

- Memory tests for Linux
  - [x] Scan-build report
  - [x] Coverity
  - [x] Valgrind (memcheck and descriptor leaks check)

- [x] Retrocompatibility with older Wazuh versions
- [x] Added unit tests (for new features)

### CODE REVIEW

File: src/remoted/manager.c [hunk 1]
```diff
@@ -1070,6 +975,194 @@ STATIC void process_deleted_multi_groups() {
...
+                    closedir(dir);
+                    continue;
+                }
+            }
+
+            copy_directory(new_src_path, new_dst_path, group);
+            closedir(dir);
+        }
+    }
+    free_strarray(files);
```
Comment (TomasTurina) [inline src/remoted/manager.c:1163]: Not necessary, but add a return like you did in the previous function to indicate the end of the function.

File: src/unit_tests/remoted/test_manager.c [hunk 2]
```diff
@@ -2411,7 +2118,1225 @@ void test_c_files(void **state)
...
+
+    free_file_sum(f_sum);
+}
+
+void test_validate_shared_files_max_path_size(void **state)
+{
+    file_sum **f_sum = NULL;
+    unsigned int f_size = 0;
+    os_calloc(2, sizeof(file_sum *), f_sum);
+    const char *base_path = "190-characters-aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa";
```
Comment (cborla) [inline src/unit_tests/remoted/test_manager.c:2131]: The 190 characters string could be a define out of the function to be use for all functions that have to.

**Review by cborla — DISMISSED** [review]
LGTM

**Review by TomasTurina — DISMISSED** [review]

**Review by cborla — APPROVED** [review]
LGTM

**Review by TomasTurina — APPROVED** [review]

**Review by vikman90 — APPROVED** [review]

**Pull request merged**